cli/src/pcluster/models/imagebuilder.py

```diff
@@ -150,4 +170,23 @@ def __init__(
         self.image = image
         self.build = build
         self.dev_settings = dev_settings
-        self._add_validator(BaseAMIValidator, ami_id=self.build.parent_image)
+        self._set_default()
+        self._add_validator(
+            EbsVolumeTypeSizeValidator, priority=10, volume_type=Param("gp2"), volume_size=self.image.root_volume.size
+        )
+
+    def _set_default(self):
+        # set default root volume
+        if self.image.root_volume is None or self.image.root_volume.size.value is None:
```

Comment:
This use case is very interesting because we're setting a default by using the values coming from other fields. 

I think this logic should stay in the Volume because it's the responsible to set the default but it's not yet clear to me how to propagate the required info from the build.parent_image.

Let's add a FIXME for now and let's think to some alternatives.